src/jit_channel/channel_manager.rs

```diff
@@ -675,13 +712,40 @@ where
 					let mut peer_state = inner_state_lock.lock().unwrap();
 					if let Some(jit_channel) = peer_state.outbound_channels_by_scid.get_mut(&scid) {
 						match jit_channel.channel_ready() {
-							Ok((intercept_id, amt_to_forward_msat)) => {
-								self.channel_manager.forward_intercepted_htlc(
-									intercept_id,
-									channel_id,
-									*counterparty_node_id,
-									amt_to_forward_msat,
-								)?;
+							Ok((htlcs, total_amt_to_forward_msat)) => {
+								// TODO: review strategy for handling batch forwarding of these htlcs
+								let total_received_msat: u64 = htlcs
+									.iter()
+									.map(|htlc| htlc.expected_outbound_amount_msat)
+									.sum();
+
+								let mut fee_remaining_msat =
+									total_received_msat - total_amt_to_forward_msat;
+								let fee_percentage =
+									fee_remaining_msat as f64 / total_received_msat as f64;
```

Comment:
hm, how do you avoid it when working with percentages? i tried to avoid rounding errors below by always rounding up to nearest msat and then taking min of remaining amount